### ISSUE TITLE
Compatibility with Alfawise u30 pro

### ISSUE THREAD

**Issue by marcodori** [LONGER3D/Marlin1.1.9_LGT0.3.x_Alfawise_Ux0Pro#1]
### Description

I've read the configuration file and i saw that the driver for both u20 and u30 pro is the same (A4988) while the u30 pro have mounted the TMC2208 driver.

There is a complete compatibility with the product or it must be reviewed?

**Comment by LONGER3D** [comment]
Hello, firmware is a complete compatibility with the product,it is available for both A4988 and TMC2208. If your machine is U20_Pro, you only need unremark "U20_Pro" in Configuration.h

**Comment by poisonedPawn** [comment]
I agree with @marcodori, it's confusing.

Then if i have u30 pro, we must to write TMC2208 for X, Y and Z axes or just define U30_Pro machine??

**Comment by LONGER3D** [comment]
Hello， TMC2208 driver board solution can be fully compatible with the existing A988 driver module, (Note: the original A988 driver work segmentation must be in 8 subdivision or 16 subdivision, that is, MS1 and MS2 pins must be pulled up) MS1 and MS2 The pin must be pulled up and the TMC2208 driver has 256 subdivision effects. In the firmware, although the motor drive selects the A4988. the driving current waveform of the TMC2208 internal processing algorithm output to the motor is 256 subdivision. The biggest advantage of this is that it reduces the burden on the master MUC, but the power consumption and heat generation of the driver IC itself are increased. If my answer is not very accurate, please correct me, we can discuss, learn, and progress together. 
  Thank you!




------------------ 原始邮件 ------------------
发件人: "Jaime Ortiz Córdoba"<notifications@github.com>; 
发送时间: 2019年8月27日(星期二) 上午6:42
收件人: "LONGER3D/Marlin1.1.9_LGT0.3.x_Alfawise_Ux0Pro"<Marlin1.1.9_LGT0.3.x_Alfawise_Ux0Pro@noreply.github.com>; 
抄送: "懂╰☆╮Y&#x2661;_&#x2661;M"<2059212139@qq.com>; "Comment"<comment@noreply.github.com>; 
主题: Re: [LONGER3D/Marlin1.1.9_LGT0.3.x_Alfawise_Ux0Pro] Compatibilitywith Alfawise u30 pro (#1)




I agree with @marcodori, it's confusing

**Comment by DeviousPenguin** [comment]
So just to confirm, if the board has a TMC2208 driver, it is best to marlin setting at A4988?

**Comment by mrouillard** [comment]
Yes! The TMC2208 will do the rest by some kind of interpolation ;)

**Comment by DeviousPenguin** [comment]
> Yes! The TMC2208 will do the rest by some kind of interpolation ;)

Ok, thank you. :) Just wanted to be 100% sure.
